aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -41,44 +60,34 @@ Tensor to(const Tensor& self, const TensorOptions& options, bool non_blocking, b
     device_opt = ensure_has_index(device_opt.value());
   }
   const auto & dtype_opt = options.dtype_opt();
-  if ((!device_opt || self.device() == device_opt.value()) &&
-      (!dtype_opt  || self.dtype()  ==  dtype_opt.value()) && !copy) {
-    return self;
-  }
   auto specified_options = self.options();
   if (device_opt) {
     specified_options = specified_options.device(device_opt.value());
   }
   if (dtype_opt) {
     specified_options = specified_options.dtype(dtype_opt.value());
   }
-  return to_impl(self, specified_options, non_blocking);
+  return to_impl(self, specified_options, non_blocking, copy, optional_memory_format);
 }
 
-Tensor to(const Tensor& self, Device device, ScalarType dtype, bool non_blocking, bool copy) {
+Tensor to(const Tensor& self, Device device, ScalarType dtype, bool non_blocking, bool copy, c10::optional<c10::MemoryFormat> optional_memory_format) {
   device = ensure_has_index(device);
-  if (self.device() == device && self.dtype() == dtype && !copy) {
-    return self;
-  }
-  return to_impl(self, self.options().device(device).dtype(dtype), non_blocking);
+  return to_impl(
+      self,
+      self.options().device(device).dtype(dtype),
```

Comment:
I know this code originally existed, but I want to point out that it is pretty weird that we have to call `self.options()` here, because the point of TensorOptions is that it has optional fields, so `to_impl` should be the method that actually handles the population in the end. No change necessary.